### PR TITLE
example: added extended public key derivation

### PR DESCRIPTION
**Pull request by edtubbs**
doc: added extended public key derivation

### CODE REVIEW

**Review by xanimo — APPROVED** [review]
ACK, tested on x86_64-pc-linux-gnu.

File: contrib/examples/example.c [hunk 1]
```diff
@@ -249,6 +249,26 @@ int main() {
...
 
+	// EXTENDED PUBLIC KEY DERIVATION EXAMPLE
+	printf("\n\nBEGIN EXTENDED PUBLIC KEY DERIVATION:\n\n");
+
+	// Generate the Master key from a seed
+	dogecoin_hdnode_from_seed(utils_hex_to_uint8("000102030405060708090a0b0c0d0e0f"), 16, &node);
+
+	// Serialize and print the Master public key
+	char master_public_key[112] = {0};
+	dogecoin_hdnode_serialize_public(&node, &dogecoin_chainparams_main, master_public_key, sizeof(master_public_key));
```
Comment (michilumin) [inline contrib/examples/example.c:260]: Just going to keep remarking to keep it in mind: we need to do something differently than requiring memaddresses for pointers here (&node, &chainparams) - so we need to definitely say a TODO: on adding a different way to access these.  Other languages may not be able to access or manipulate pointer addresses; and we don't want to restrict. Probably for chainparams a way to select main or testnet;  node probably can be something that works out an internal reference.

**Review by michilumin — APPROVED** [review]
Ok for this phase with the standing caveat of TODO: on have to look for alternatives to pointers and mem address refs.

File: contrib/examples/example.c (hunk 1 above)
Comment (edtubbs) [reply contrib/examples/example.c:260]: Agreed, I'm working on wrappers to pass individual parameters back by reference.
https://github.com/edtubbs/libdogecoin/commit/56f08a75d6067cc5de548961a814ee58a3d66f6f#diff-b6c5c7f340ac317dd7d04288fc653a288892e2fa73389a9ed8a073328a3858e7R237-R256

Comment (edtubbs) [reply contrib/examples/example.c:260]: Resolved in #158.

**Comment by edtubbs** [comment]
Closing due to #158

**Pull request closed without merging**